-------------------------------------------------------------------------------
Test set: fr.ensea.rts.kardellas.suprano.server.TCPServerTest
-------------------------------------------------------------------------------
Tests run: 3, Failures: 1, Errors: 1, Skipped: 0, Time elapsed: 0.052 s <<< FAILURE! -- in fr.ensea.rts.kardellas.suprano.server.TCPServerTest
fr.ensea.rts.kardellas.suprano.server.TCPServerTest.testHandleConnection -- Time elapsed: 0.011 s <<< FAILURE!
org.opentest4j.AssertionFailedError: Unexpected exception thrown: java.lang.NullPointerException: Cannot invoke "java.net.ServerSocket.accept()" because "<local2>.serverSocket" is null
	at org.junit.jupiter.api.AssertionFailureBuilder.build(AssertionFailureBuilder.java:152)
	at org.junit.jupiter.api.AssertDoesNotThrow.createAssertionFailedError(AssertDoesNotThrow.java:84)
	at org.junit.jupiter.api.AssertDoesNotThrow.assertDoesNotThrow(AssertDoesNotThrow.java:53)
	at org.junit.jupiter.api.AssertDoesNotThrow.assertDoesNotThrow(AssertDoesNotThrow.java:36)
	at org.junit.jupiter.api.Assertions.assertDoesNotThrow(Assertions.java:3199)
	at fr.ensea.rts.kardellas.suprano.server.TCPServerTest.testHandleConnection(TCPServerTest.java:54)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
Caused by: java.lang.NullPointerException: Cannot invoke "java.net.ServerSocket.accept()" because "<local2>.serverSocket" is null
	at fr.ensea.rts.kardellas.suprano.server.TCPServer.handleConnection(TCPServer.java:32)
	at fr.ensea.rts.kardellas.suprano.server.TCPServer.handleConnection(TCPServer.java:31)
	at fr.ensea.rts.kardellas.suprano.server.TCPServerTest.lambda$testHandleConnection$0(TCPServerTest.java:54)
	at org.junit.jupiter.api.AssertDoesNotThrow.assertDoesNotThrow(AssertDoesNotThrow.java:49)
	... 6 more

fr.ensea.rts.kardellas.suprano.server.TCPServerTest.testLaunch -- Time elapsed: 0.006 s <<< ERROR!
java.lang.RuntimeException: Failed to launch server
	at fr.ensea.rts.kardellas.suprano.server.TCPServer.launch(TCPServer.java:24)
	at fr.ensea.rts.kardellas.suprano.server.TCPServer.launch(TCPServer.java:20)
	at fr.ensea.rts.kardellas.suprano.server.TCPServerTest.testLaunch(TCPServerTest.java:42)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
Caused by: java.net.BindException: Address already in use: bind
	at java.base/sun.nio.ch.Net.bind0(Native Method)
	at java.base/sun.nio.ch.Net.bind(Net.java:565)
	at java.base/sun.nio.ch.Net.bind(Net.java:554)
	at java.base/sun.nio.ch.NioSocketImpl.bind(NioSocketImpl.java:636)
	at java.base/java.net.ServerSocket.bind(ServerSocket.java:391)
	at java.base/java.net.ServerSocket.<init>(ServerSocket.java:278)
	at java.base/java.net.ServerSocket.<init>(ServerSocket.java:171)
	at fr.ensea.rts.kardellas.suprano.server.TCPServer.launch(TCPServer.java:20)
	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:733)
	at org.mockito.internal.util.reflection.InstrumentationMemberAccessor$Dispatcher$ByteBuddy$Ej8oXWFM.invokeWithArguments(Unknown Source)
	at org.mockito.internal.util.reflection.InstrumentationMemberAccessor.invoke(InstrumentationMemberAccessor.java:251)
	at org.mockito.internal.util.reflection.ModuleMemberAccessor.invoke(ModuleMemberAccessor.java:55)
	at org.mockito.internal.creation.bytebuddy.MockMethodAdvice.tryInvoke(MockMethodAdvice.java:314)
	at org.mockito.internal.creation.bytebuddy.MockMethodAdvice$RealMethodCall.invoke(MockMethodAdvice.java:234)
	at org.mockito.internal.invocation.InterceptedInvocation.callRealMethod(InterceptedInvocation.java:142)
	at org.mockito.internal.stubbing.answers.CallsRealMethods.answer(CallsRealMethods.java:45)
	at org.mockito.Answers.answer(Answers.java:90)
	at org.mockito.internal.handler.MockHandlerImpl.handle(MockHandlerImpl.java:111)
	at org.mockito.internal.handler.NullResultGuardian.handle(NullResultGuardian.java:29)
	at org.mockito.internal.handler.InvocationNotifierHandler.handle(InvocationNotifierHandler.java:34)
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:82)
	at org.mockito.internal.creation.bytebuddy.MockMethodAdvice.handle(MockMethodAdvice.java:134)
	... 5 more

